Code/Framework/AzCore/AzCore/Memory/AllocatorBase.cpp

```diff
@@ -11,10 +11,157 @@
 
 using namespace AZ;
 
-AllocatorBase::AllocatorBase(IAllocatorAllocate* allocationSource, const char* name, const char* desc) :
-    IAllocator(allocationSource),
-    m_name(name),
-    m_desc(desc)
+#define RECORDING_ENABLED 0
```

Comment:
Can you get rid of the `using namespace AZ` in this file?
Also it seems like RECORD_ENABLED could be a common macro name, that could be part of some kind video/audio authoring 3rdParty.
How about `O3DE_RECORDING_ENABLED`?

Finally it seems like we should only define it to `0` if it is not defined already.

```
#if !defined(RECORDING_ENABLED)
    #define RECORDING_ENABLED 0
#endif
```

Reply:
will rename to `O3DE_MEMORY_RECORDING_ENABLED`, its just a way to get recordings, pretty manual, so I wouldnt expect someone to pass it to this compilation unit (its just a something to enable to get benchmark data out)